chips/sam4l/src/usart.rs

```diff
@@ -368,12 +580,12 @@ impl USART {
     }
 
     fn set_mode(&self, mode: u32) {
-        let regs: &USARTRegisters = unsafe { &*self.registers };
+        let regs: &UsartRegisters = unsafe { &*self.registers };
```

Comment:
I think the right choice here is to change the type of `mode` to accept a combination of bits. (Although, I confess to not having any idea what that type is. Just try putting a combination of bits in a call to `set_mode` and the compiler will tell you.)